doc/api.rst

```diff
@@ -49,7 +49,7 @@ Three types of boundary condition are currently supported:
 Name             Description
 ===============  =========================================
 GO_BC_NONE       No boundary conditions are applied.
-GO_BC_EXTERNAL   Some external forcing is applied. This must be implemented by a kernel. The domain must be defined with a T-point mask (see :ref:`gocean1.0-grid-init`).
+GO_BC_EXTERNAL   Some external forcing is applied. This must be implemented by a kernel.
```

Comment:
The documentation in grid_mod.f90 states:
>  !! ... However, we also support the
     !! specification of a model consisting entirely of wet points

But that never worked, so I've fixed that.

Ah .... I just went back to the original commit, and realised it also had "... with PBC" :(  So, not really a fix as I thought. 
I would prefer not to require a tmask, it is really a small change (the rest works), leaving it up to the user how to setup the boundary condition. E.g. the test cases could all be simplified by removing the tmask code (it's just set to '1'), see my new test case.
Am happy to discuss if this should really be required (though it will make code easier if the need for tmask is just removed - to me feels the library is more generic this way).

Reply:
MPI version of PSyclone-generated nemolite2d...
1 process:
ua checksum =   0.52670513E+02
va checksum =   0.57100321E+03
2 processes:
ua checksum =   0.52670513E+02
va checksum =   0.57100321E+03
4 processes:
ua checksum =   0.52670513E+02
va checksum =   0.57100321E+03
...so that's all good.

Reply:
The point is that nemolite2d in PSycloneBench *does* use the T-mask to define an arbitrary domain. This is stored within the infrastructure so it can be supplied (in the PSy layer) to those kernels that require it, Therefore, I don't think we can remove it.

Reply:
Therefore, it is really required. The reason PBCs aren't implemented is because we never got as far as 'PSyKAl'ing the 'shallow' benchmark which uses them - nemolite2d does not.

Reply:
All in all, I think we need to revert this change!

Reply:
In our telco we agreed that we should modify the test for PBC to abort if we are in an unsupported configutation if no tmask is specified (i.e. abort in distributed memory, but allow single process, since this is what e.g. the shallow benchmark uses).

If no tmask is specified, we add a dummy tmask (set to all wet) based on the specified grid dimension. This way any kernel that requests a tmask will get valid data.

I've also reworked the documentation to reflect this.